File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/order/OrderSupplier.kt
package com.sudo248.sudoo.domain.entity.order

import com.sudo248.base_android.base.ItemDiff
import com.sudo248.sudoo.domain.entity.discovery.SupplierInfo
import com.sudo248.sudoo.domain.entity.promotion.Promotion
import java.time.LocalDateTime

data class OrderSupplier(
    val orderSupplierId: String,
    val supplier: SupplierInfo,
    val promotion: Promotion?,
    val status: OrderStatus,
    val shipment: Shipment,
    val totalPrice: Double,
    val expectedReceiveDateTime: LocalDateTime,
    val orderCartProducts: List<OrderCartProduct>
) : ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplier && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplier && other.orderSupplierId == orderSupplierId
                && other.supplier.supplierId == this.supplier.supplierId
                && orderCartProducts.size == other.orderCartProducts.size
    }

}

File: client/web/sudoo/lib/domain/model/order/order_supplier.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/discovery/supplier.dart';
import 'package:sudoo/domain/model/order/order_status.dart';
import 'package:sudoo/domain/model/order/shipment.dart';
import 'package:sudoo/domain/model/promotion/promotion_info.dart';

import '../discovery/order_cart_product.dart';

part 'order_supplier.g.dart';

@JsonSerializable(explicitToJson: true)
class OrderSupplier {
  final String orderSupplierId;
  final Supplier supplier;
  final PromotionInfo? promotion;
  final double totalPrice;
  final Shipment shipment;
  final DateTime expectedReceiveDateTime;
  OrderStatus status;
  final List<OrderCartProduct> orderCartProducts;

  OrderSupplier(
    this.orderSupplierId,
    this.supplier,
    this.promotion,
    this.totalPrice,
    this.shipment,
    this.expectedReceiveDateTime,
    this.status,
    this.orderCartProducts,
  );

  factory OrderSupplier.fromJson(Map<String, dynamic> json) => _$OrderSupplierFromJson(json);

  Map<String, dynamic> toJson() => _$OrderSupplierToJson(this);
}
